File: BE/alfredo/src/main/java/org/b104/alfredo/user/Dto/UserUpdateDto.java
package org.b104.alfredo.user.Dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.b104.alfredo.user.Domain.User;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class UserUpdateDto {
    private String nickname;
    private Date birth;
    private List<Long> answer;
    private String googleCalendarUrl;

    public static void updateEntity(User user, UserUpdateDto userUpdateDto) {
        if (userUpdateDto.getNickname() != null) user.setNickname(userUpdateDto.getNickname());
        if (userUpdateDto.getBirth() != null) user.setBirth(userUpdateDto.getBirth());
        if (userUpdateDto.getAnswer() != null) user.setAnswer(userUpdateDto.getAnswer());
        if (userUpdateDto.getGoogleCalendarUrl() != null) user.setGoogleCalendarUrl(userUpdateDto.getGoogleCalendarUrl());
    }
}

File: FE/alfredo/lib/models/user/user_update_dto.dart
class UserUpdateDto {
  String? nickname;
  DateTime? birth;
  String? answer;
  String? googleCalendarUrl;

  UserUpdateDto(
      {this.nickname, this.birth, this.answer, this.googleCalendarUrl});

  Map<String, dynamic> toJson() {
    Map<String, dynamic> data = {};
    if (nickname != null) data['nickname'] = nickname;
    if (birth != null) data['birth'] = birth!.toIso8601String();
    if (answer != null) data['answer'] = answer;
    if (googleCalendarUrl != null)
      data['googleCalendarUrl'] = googleCalendarUrl;
    return data;
  }
}
